pkgs/development/libraries/alice-vision/3.0.0.nix

```diff
@@ -0,0 +1,225 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, pkg-config
+
+, pkgs
+
+, assimp
+# use the same boost as in cctag here
+, boost179
+, ceres-solver
+, clp
+, eigen
+, expat
+, flann
+, geogram
+, lemon-graph
+, lz4
+, openexr
+, openimageio
+, zlib
+
+, enableOpenMP ? true
+# Separate openmp input only required on Darwin stdenv (LLVM)
+, openmp
+
+, enableAlembic ? true, alembic
+, enableCctag ? true, cctag
+, enableOpenCV ? true, enableOpenCVContrib ? enableOpenCV, opencv
+}:
+
+let
+  nanoflann = pkgs.nanoflann.overrideAttrs (
+    oldAttrs: rec {
+      version = "1.4.2";
+      src = fetchFromGitHub {
+        owner = "jlblancoc";
+        repo = "nanoflann";
+        rev = "v${version}";
+        hash = "sha256-znIX1S0mfOqLYPIcyVziUM1asBjENPEAdafLud1CfFI=";
+      };
+    }
+  );
+in
+stdenv.mkDerivation rec {
+  pname = "alice-vision";
+  version = "3.0.0";
+
+  outputs = [ "out" "dev" ];
+
+  src = fetchFromGitHub {
+    owner = "alicevision";
+    repo = "AliceVision";
+    rev = "v${version}";
+    hash = "sha256-rFd2AFtvC1RugKGv4tI2k3rtDqXHjdDC3pgOrqDqIT0=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    flann
+    nanoflann
+    openexr
+
+    # Temporary fix until flann 1.9.2 is in Nixpkgs
+    lz4
+  ];
+
+  propagatedBuildInputs = [
+    assimp
+    boost179
+    ceres-solver
+    clp
+    eigen
+    expat
+    geogram
+    lemon-graph
+    openimageio
+    zlib
+  ] ++ lib.optional enableAlembic alembic
+    ++ lib.optional enableCctag cctag
+    ++ lib.optional enableOpenCV opencv
+    ++ lib.optional stdenv.cc.isClang openmp;
+
+  patches = [
+    # Don't optimize for the host processor (could lead to reproducibility issues)
+    ./cmake-disable-optimize-for-architecture.patch
+
+    # Upstream uses patched Clp/Osi/CoinUtils containing CMake build scripts.
+    # Instead of the patched versions, we re-use our packages and add CMake
+    # find modules that use the pkg-config files already generated by those
+    # dependencies.
+    ./find-coin-modules.patch
+
+    ./find-flann.patch
+  ];
+
+  # Instead of using dependencies from Git submodules, we use Nix packages
+  # This speeds up fetching and reduces the source archive size
+  postPatch = ''
+    rmdir src/dependencies/nanoflann
+    ln -s ${nanoflann} src/dependencies/nanoflann
```

Comment:
It uses the source code or the compiled artifacts?

Reply:
`nanoflann` is header-only, so no compiled artifacts to use.